streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -749,6 +754,11 @@ private Map<TopicPartition, Long> committedOffsetForChangelogs(final Map<TaskId,
         }
     }
 
+    private void filterNewPartitionsToRestore(final Map<TaskId, Task> tasks, final Set<ChangelogMetadata> newPartitionsToRestore) {
```

Comment:
This is to fix the issue found before: if a task is paused, we should not initialize their changelogs, and hence the restore consumer would not poll their records, and hence we would never got a task to be restore while it's not in the `tasks` set.